src/lib/httpd.ts

```diff
@@ -0,0 +1,161 @@
+import { derived, get, writable } from "svelte/store";
+
+import { HttpdClient } from "@httpd-client";
+
+export interface StoredSession {
+  id: string;
+  publicKey: string;
+}
+
+export type HttpdState =
+  | { state: "stopped" }
+  | { state: "running" }
+  | {
+      state: "authenticated";
+      session: StoredSession;
+    };
+
+const LOCAL_STORAGE_KEY = "httpd";
+
+const store = writable<HttpdState | undefined>(undefined);
```

Comment:
It seems like we don't really need the `undefined` state:

```diff
diff --git a/src/App/Header/Connect.svelte b/src/App/Header/Connect.svelte
index 7101f1b4..37e4ccf6 100644
--- a/src/App/Header/Connect.svelte
+++ b/src/App/Header/Connect.svelte
@@ -103,11 +103,11 @@
 <Floating>
   <div slot="toggle">
     <Button
-      style={$httpdStore?.state === "authenticated"
+      style={$httpdStore.state === "authenticated"
         ? "padding-left: 10px; padding-right: 1rem;"
         : undefined}
       variant="foreground">
-      {#if $httpdStore?.state === "authenticated"}
+      {#if $httpdStore.state === "authenticated"}
         <div class="toggle-avatar">
           <div style:height="1.5rem">
             <Avatar nodeId={$httpdStore.session.publicKey} />
@@ -123,7 +123,7 @@
   </div>
 
   <div slot="modal">
-    {#if $httpdStore?.state === "authenticated"}
+    {#if $httpdStore.state === "authenticated"}
       <div class="dropdown">
         <div class="avatar-id-container">
           <div class="dropdown-avatar">
@@ -147,7 +147,7 @@
           Disconnect
         </div>
       </div>
-    {:else if $httpdStore?.state === "running"}
+    {:else if $httpdStore.state === "running"}
       <div class="dropdown info">
         To connect to your local Radicle node, run this command in your
         terminal:
diff --git a/src/components/Thread.svelte b/src/components/Thread.svelte
index 8d7569b5..1e3b7895 100644
--- a/src/components/Thread.svelte
+++ b/src/components/Thread.svelte
@@ -70,7 +70,7 @@
     authorAlias={root.author.alias}
     timestamp={root.timestamp}
     body={root.body}
-    showReplyIcon={Boolean($httpdStore)}
+    showReplyIcon={$httpdStore.state === "authenticated"}
     on:toggleReply={toggleReply} />
 </div>
 {#each replies as reply}
diff --git a/src/lib/httpd.ts b/src/lib/httpd.ts
index ca261bb0..20dd9a0a 100644
--- a/src/lib/httpd.ts
+++ b/src/lib/httpd.ts
@@ -17,7 +17,7 @@ export type HttpdState =
 
 const LOCAL_STORAGE_KEY = "httpd";
 
-const store = writable<HttpdState | undefined>(undefined);
+const store = writable<HttpdState>({ state: "stopped" });
 export const httpdStore = derived(store, s => s);
 
 const api = new HttpdClient({
@@ -56,7 +56,7 @@ function pollSession() {
     const httpd = get(httpdStore);
 
     // Checking the current state of the httpd instance
-    if (httpd?.state !== "authenticated") {
+    if (httpd.state !== "authenticated") {
       try {
         await api.getNode();
         store.set({ state: "running" });
@@ -72,7 +72,7 @@ function pollSession() {
           sess.status === "unauthorized" ||
           sess.expiresAt < unixTimeInSeconds
         ) {
-          clear({ state: "stopped" });
+          clear({ state: "running" });
         }
       } catch {
         clear({ state: "stopped" });
@@ -83,7 +83,7 @@ function pollSession() {
 
 export async function disconnect() {
   const httpd = get(store);
-  if (httpd?.state !== "authenticated") {
+  if (httpd.state !== "authenticated") {
     return "success";
   }
   await api.session.delete(httpd.session.id);
@@ -108,7 +108,7 @@ export function initialize() {
   addEventListener("storage", event => {
     if (event.key === LOCAL_STORAGE_KEY) {
       if (event.newValue === null) {
-        store.set(undefined);
+        store.set({ state: "stopped" });
       } else {
         const parsed = JSON.parse(event.newValue);
 
@@ -123,7 +123,7 @@ export function initialize() {
   });
 
   const httpd = get(store);
-  if (httpd?.state !== "authenticated") {
+  if (httpd.state !== "authenticated") {
     api
       .getNode()
       .then(() => {
diff --git a/src/views/projects/Issue.svelte b/src/views/projects/Issue.svelte
index 51e74244..69b132e2 100644
--- a/src/views/projects/Issue.svelte
+++ b/src/views/projects/Issue.svelte
@@ -30,7 +30,7 @@
   const api = new HttpdClient(baseUrl);
 
   const action: "create" | "edit" | "view" =
-    $httpdStore?.state === "authenticated" && utils.isLocal(baseUrl.hostname)
+    $httpdStore.state === "authenticated" && utils.isLocal(baseUrl.hostname)
       ? "edit"
       : "view";
   const items: Item<IssueState>[] = [
@@ -52,10 +52,7 @@
   async function createReply({
     detail: reply,
   }: CustomEvent<{ id: string; body: string }>) {
-    if (
-      $httpdStore?.state === "authenticated" &&
-      reply.body.trim().length > 0
-    ) {
+    if ($httpdStore.state === "authenticated" && reply.body.trim().length > 0) {
       await api.project.updateIssue(
         projectId,
         issue.id,
@@ -70,7 +67,7 @@
   }
 
   async function createComment(body: string) {
-    if ($httpdStore?.state === "authenticated" && body.trim().length > 0) {
+    if ($httpdStore.state === "authenticated" && body.trim().length > 0) {
       await api.project.updateIssue(
         projectId,
         issue.id,
@@ -83,7 +80,7 @@
 
   async function editTitle({ detail: title }: CustomEvent<string>) {
     if (
-      $httpdStore?.state === "authenticated" &&
+      $httpdStore.state === "authenticated" &&
       title.trim().length > 0 &&
       title !== issue.title
     ) {
@@ -101,7 +98,7 @@
   }
 
   async function saveTags({ detail: tags }: CustomEvent<string[]>) {
-    if ($httpdStore?.state === "authenticated") {
+    if ($httpdStore.state === "authenticated") {
       const { add, remove } = utils.createAddRemoveArrays(issue.tags, tags);
       if (add.length === 0 && remove.length === 0) {
         return;
@@ -121,7 +118,7 @@
   }
 
   async function saveAssignees({ detail: assignees }: CustomEvent<string[]>) {
-    if ($httpdStore?.state === "authenticated") {
+    if ($httpdStore.state === "authenticated") {
       const { add, remove } = utils.createAddRemoveArrays(
         issue.assignees,
         assignees,
@@ -144,7 +141,7 @@
   }
 
   async function saveStatus({ detail: state }: CustomEvent<IssueState>) {
-    if ($httpdStore?.state === "authenticated") {
+    if ($httpdStore.state === "authenticated") {
       await api.project.updateIssue(
         projectId,
         issue.id,
@@ -273,7 +270,7 @@
         <Thread {thread} {rawPath} on:reply={createReply} />
       {/each}
       <div style:margin-top="1rem">
-        {#if $httpdStore}
+        {#if $httpdStore.state === "authenticated"}
           <Textarea
             resizable
             on:submit={() => {
diff --git a/src/views/projects/Issue/New.svelte b/src/views/projects/Issue/New.svelte
index c4b82566..f25dbd67 100644
--- a/src/views/projects/Issue/New.svelte
+++ b/src/views/projects/Issue/New.svelte
@@ -28,7 +28,9 @@
   let preview: boolean = false;
   let action: "create" | "view";
   $: action =
-    $httpdStore && utils.isLocal(baseUrl.hostname) && !preview
+    $httpdStore.state === "authenticated" &&
+    utils.isLocal(baseUrl.hostname) &&
+    !preview
       ? "create"
       : "view";
 
diff --git a/src/views/projects/Issues.svelte b/src/views/projects/Issues.svelte
index aa86de56..977b1098 100644
--- a/src/views/projects/Issues.svelte
+++ b/src/views/projects/Issues.svelte
@@ -137,7 +137,7 @@
         {/each}
       </div>
     </div>
-    {#if $httpdStore?.state === "authenticated" && utils.isLocal(baseUrl.hostname)}
+    {#if $httpdStore.state === "authenticated" && utils.isLocal(baseUrl.hostname)}
       <ProjectLink
         projectParams={{
           view: {
diff --git a/src/views/projects/Patch.svelte b/src/views/projects/Patch.svelte
index 11f104ab..d9f21d0c 100644
--- a/src/views/projects/Patch.svelte
+++ b/src/views/projects/Patch.svelte
@@ -75,10 +75,7 @@
   async function createReply({
     detail: reply,
   }: CustomEvent<{ id: string; body: string }>) {
-    if (
-      $httpdStore?.state === "authenticated" &&
-      reply.body.trim().length > 0
-    ) {
+    if ($httpdStore.state === "authenticated" && reply.body.trim().length > 0) {
       await api.project.updatePatch(
         projectId,
         patch.id,
@@ -111,7 +108,7 @@
   }
 
   async function saveTags({ detail: tags }: CustomEvent<string[]>) {
-    if ($httpdStore?.state === "authenticated") {
+    if ($httpdStore.state === "authenticated") {
       const { add, remove } = utils.createAddRemoveArrays(patch.tags, tags);
       if (add.length === 0 && remove.length === 0) {
         return;
@@ -127,7 +124,9 @@
   }
 
   const action: "create" | "edit" | "view" =
-    $httpdStore && utils.isLocal(baseUrl.hostname) ? "edit" : "view";
+    $httpdStore.state === "authenticated" && utils.isLocal(baseUrl.hostname)
+      ? "edit"
+      : "view";
   const options = ["activity", "commits", "files"].map(o => ({
     value: o,
     title: capitalize(o),
diff --git a/src/views/projects/View.svelte b/src/views/projects/View.svelte
index a54537a6..43717bc2 100644
--- a/src/views/projects/View.svelte
+++ b/src/views/projects/View.svelte
@@ -222,7 +222,7 @@
           </div>
         {/await}
       {:else if activeRoute.params.view.resource === "issues" && activeRoute.params.view.params?.view.resource === "new"}
-        {#if $httpdStore?.state === "authenticated"}
+        {#if $httpdStore.state === "authenticated"}
           <NewIssue
             on:create={handleIssueCreation}
             session={$httpdStore.session}

```

Reply:
> It seems like we don't really need the `undefined` state:

Sure we can do that, I wanted initially define between we don't know the current state and `{status: "stopped"}`